azureeyemodule/app/model/ssd.cpp

```diff
@@ -55,14 +87,30 @@ void SSDModel::run(cv::GStreamingCompiled *pipeline)
 
         // Log some metadata.
         this->log_parameters();
+        bool ran_out_naturally;
+
+        if (this->inputsource == "uvc" || this->inputsource.rfind(this->VIDEO_PREFIX, 0) == 0 ) 
```

Comment:
Two things:
1. Please put parentheses around the different conditionals in this if statement, just because it is easier to read and better practice (since C often has surprising operator precedence rules).
2. This whole if statement is the type of thing I would love to avoid by combining the `compile_cv_graph_uvc_video` with `compile_cv_graph`. I feel like it should be not too hard, since the only thing that should really change is the input source. Then you could also combine `pull_data_uvc_video` with `pull_data`.

Reply:
1. done
2. There are quite a lot differences between 2 functions, for example, img and img_ts desync, kernel containing custom G-API op, input source. That's why I made them into 2 separate functions. Any suggestions?